api/handlers.go

```diff
@@ -26,3 +32,94 @@ func NewServer(ctx context.Context, storage interfaces.StorageFetcher, config co
 	logger := belogger.FromContext(ctx)
 	return &Server{storage: storage, logger: logger, config: config}
 }
+
+func (s Server) ObjectLifeline(ctx echo.Context, objectReference string, params ObjectLifelineParams) error {
+	limit, offset, err := checkLimitOffset(params.Limit, params.Offset)
+	if err != nil {
+		return ctx.JSON(http.StatusBadRequest, NewSingleMessageError(err.Error()))
+	}
+
+	ref, errMsg := checkReference(objectReference)
+	if errMsg != nil {
+		return ctx.JSON(http.StatusBadRequest, *errMsg)
+	}
+
+	sort := "desc"
+	if params.Sort != nil {
+		s := *params.Sort
+		if s != "desc" && s != "asc" {
+			return ctx.JSON(http.StatusBadRequest, NewSingleMessageError("query parameter 'sort' should be 'desc' or 'asc'"))
+		}
+		sort = s
+	}
+
+	records, count, err := s.storage.GetLifeline(
+		ref.Bytes(),
+		params.FromIndex,
+		params.PulseNumberLt, params.PulseNumberGt,
+		limit, offset,
+		sort,
+	)
+	if err != nil {
+		errMsg := err.Error()
+		if strings.Contains(errMsg, "query parameter") {
+			return ctx.JSON(http.StatusBadRequest, NewSingleMessageError(errMsg))
+
+		}
+		s.logger.Error(err)
```

Comment:
could you pls log above? before IF statement

Reply:
Its not needed above, because we should log only InternalServerErrors, and above we get error from users input (query parameters), its not strange situation.